### PR TITLE
added default firewall to leader if getting firewalls fails

### PR DESCRIPTION
**Pull request by ashleyschuett**

### CODE REVIEW

**Comment by nicktate** [comment]
@normanjoyner suggested ashley do this in case some part of the provision process fails the bed so we can still ssh to the cluster.

File: lib/leader.js [hunk 1]
```diff
@@ -282,6 +292,24 @@ module.exports = {
             }
         }
 
+        function set_default_firewall(callback) {
```
Comment (normanjoyner) [inline lib/leader.js]: I don't think this needs to happen here. I would rather firewall specific logic be implemented in `lib/firewall.js`. fallback default rule can be added to the `get_cloud_rules` function

File: lib/leader.js [hunk 2]
```diff
@@ -174,9 +174,19 @@ module.exports = {
                 request(options, (err, response) => {
                     if(err) {
                         core.loggers['containership-cloud'].log('warn', `Unable to fetch firewalls from ContainerShip Cloud: ${err.message}`);
+
+                        if(!cache.firewalls) {
```
Comment (normanjoyner) [inline lib/leader.js]: not needed

File: lib/leader.js [hunk 3]
```diff
@@ -174,9 +174,19 @@ module.exports = {
...
+                            return set_default_firewall(callback);
+                        }
+
                         return callback();
                     } else if(response.statusCode !== 200) {
                         core.loggers['containership-cloud'].log('warn', `Unable to fetch firewalls from ContainerShip Cloud: API returned ${response.statusCode}.`);
+
+                        if(!cache.firewalls) {
```
Comment (normanjoyner) [inline lib/leader.js]: not needed

File: lib/firewall.js [hunk 4]
```diff
@@ -155,8 +155,14 @@ class Firewall {
 
     get_cloud_rules(callback) {
         this.core.cluster.myriad.persistence.get(constants.myriad.FIREWALLS, (err, firewalls) => {
+            const default_firewall = [{
```
Comment (normanjoyner) [inline lib/firewall.js]: going to nitpick: would prefer a plural variable such as `default_firewall_rules`

**Pull request merged**